### PR TITLE
WIP add support for EIP1559 and CIP42 to ContractKit (wallet-base)

### LINKED ISSUE #10447: Add CIP42 support to ContractKit / SDKs
### Expected Behavior

can send and parse transactions of the type defined in https://github.com/celo-org/celo-proposals/blob/master/CIPs/cip-0042.md

### Current Behavior

only legacy transactions supoorted

## see also 
#10448

### PR DESCRIPTION
**Pull request by aaronmgdr**
### Description

This WIP Change adds eip1559 and cip42 signing / serialization. recovery to Contractkit (actually to wallet-base and wallet-local and celo/connect, as these are were the actions happens)

### Other changes

_Describe any minor or "drive-by" changes here._

### Tested

New and Old tests

### Related issues

- Fixes #10447 

### Backwards compatibility

No. Deprecated Functions related to gasPrice have been removed.

### CODE REVIEW

File: packages/sdk/wallets/wallet-base/src/signing-utils.ts [hunk 1]
```diff
@@ -171,34 +379,173 @@ export function extractSignature(rawTx: string): { v: number; r: Buffer; s: Buff
...
+    throw new Error(e)
+  }
+}
+
+export function getSignerFromTx(serializedTransaction: string): string {
+  const transactionArray: any[] = RLP.decode(`0x${serializedTransaction.slice(4)}`)
+  const signer = getPublicKeyofSignerFromTx(transactionArray)
+  const address = publicToAddress(signer)
+  return `0x` + address.toString('hex')
+}
```
Comment (aaronmgdr) [inline packages/sdk/wallets/wallet-base/src/signing-utils.ts:448]: this doesnt work

File: .github/workflows/circleci.yml
Comment (soloseng) [inline .github/workflows/circleci.yml:1]: This should not be changed.

It is currently preventing GH action from running.

Comment (aaronmgdr) [reply .github/workflows/circleci.yml:1]: yeah when i started i branched off the node 18 branch. im gonna try moving my commits to a new branch off master

**Comment by aaronmgdr** [comment]
closed for https://github.com/celo-org/celo-monorepo/pull/10483

**Pull request closed without merging**